scripts/pem_to_pub_c.py

```diff
@@ -29,33 +26,38 @@ def main():
 
     args = get_args()
 
-    f = open(args.key, 'r')
-    key = RSA.importKey(f.read())
-    f.close
-
-    f = open(args.out, 'w')
-
-    f.write("#include <stdint.h>\n")
-    f.write("#include <stddef.h>\n\n")
-
-    f.write("const uint32_t " + args.prefix + "_exponent = " +
-            str(key.publickey().e) + ";\n\n")
-
-    f.write("const uint8_t " + args.prefix + "_modulus[] = {\n")
-    i = 0
-    for x in array.array("B", long_to_bytes(key.publickey().n)):
-        f.write("0x" + '{0:02x}'.format(x) + ",")
-        i = i + 1
-        if i % 8 == 0:
-            f.write("\n")
-        else:
-            f.write(" ")
-    f.write("};\n")
-
-    f.write("const size_t " + args.prefix + "_modulus_size = sizeof(" +
-            args.prefix + "_modulus);\n")
-
-    f.close()
+    with open(args.key, 'r') as f:
+        key = RSA.importKey(f.read())
+
+    with open(args.out, 'w') as f:
+        f.write("#include <stdint.h>\n")
+        f.write("#include <stddef.h>\n\n")
+
+        # Refuse public exponents with more than 32 bits. Otherwise the C
+        # compiler will just give an overflow warning and truncate.
+        # This will lead to TAs seemingly having invalid signatures with a
+        # possible security issue for any e = k*2^32 + 1 (for any integer k).
+        if key.publickey().e > 4294967295:
```

Comment:
`0xffffffff` or some other defined constant to make it more readable for programmers? Also I wouldn't mind adding an `and key.publickey().e < 3` to this if-check.

Reply:
@jforissier pep8 gave me a few issues (mostly code formatting), so I fixed that along the way.

@jbech-linaro Changed the constant to hex. I did not add the other check: The raised error message would be wrong and int would not match the intention of the check. The check is not to weed out bad/weak/invalid RSA keys but for the case where a valid and perfectly fine RSA key is provided, but not supported by OP-TEE (for efficiency reasons).